test/rules/no-unbound-method/default/test.ts.lint

```diff
@@ -1,3 +1,18 @@
+class C {
+    constructor() {
+        this.foo = 1;
+        this.methodA = this.methodA.bind(this);
+    }
+    methodA() {}
+    methodB() {}
+}
+
+const c = new C();
+// OK if bound in constructor
+[0].forEach(c.methodA);
+[0].forEach(c.methodB);
+            ~~~~~~~~~ [0]
+
```

Comment:
There should be another test:
```ts
class D extends C {
    methodA() {}
    methodB() {}
}
const d = new D();
[0].forEach(d.methodA);
[0].forEach(d.methodB);
            ~~~~~~~~~[0]
```

I think this will not work with the current implementation.